src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -86,5 +89,33 @@ private class IsolatedRSessionEvaluation : IRSessionEvaluation {
                 _session.StopHostAsync().DoNotWait();
             }
         }
+
+        public bool IsRemote => BrokerConnector.IsRemote;
+        public Uri BrokerUri => BrokerConnector.BrokerUri;
+
+        public Task<RHost> ConnectAsync(string name, IRCallbacks callbacks, string rCommandLineArguments = null, int timeout = 3000, CancellationToken cancellationToken = new CancellationToken())
+            => BrokerConnector.ConnectAsync(name, callbacks, rCommandLineArguments, timeout, cancellationToken);
+
+        public event EventHandler BrokerChanged {
+            add { BrokerConnector.BrokerChanged += value; }
+            remove { BrokerConnector.BrokerChanged -= value; }
+        }
+
+        public Task<bool> TrySwitchBroker(string name, string path = null) {
+            path = path ?? new RInstallation().GetRInstallPath();
+
+            Uri uri;
+            if (!Uri.TryCreate(path, UriKind.Absolute, out uri)) {
+                return Task.FromResult(false);
+            }
+
+            if (uri.IsFile) {
+                BrokerConnector.SwitchToLocalBroker(name, uri.LocalPath);
```

Comment:
Connector will be removed completely, in the next PR this method will have contain the whole implementation

---
In reply to: [77388081](https://github.com/Microsoft/RTVS/pull/2244#discussion_r77388081) [](ancestors = 77388081)